examples/llm/src/mosaic_gpt.py

```diff
@@ -525,10 +526,26 @@ def loss(self, outputs, batch):
     def get_metrics(self, is_train=False):
         return self.train_metrics if is_train else self.eval_metrics
 
-    def update_metric(self, batch, outputs, metric):
-        outputs = outputs.view(-1, outputs.size(-1))
-        targets = self.get_targets(batch).view(-1)
-        metric.update(outputs, targets)
+    def update_metric(self, batch, outputs, metric) -> None:
+        if isinstance(metric, InContextLearningMetric):
```

Comment:
Should we just raise here `if batch.get('mode', None) == 'icl_task' and not isinstance(metric, InContextLearningMetric)`?

Reply:
Punting this to future PR